### PR TITLE
Pt 155784885 aws deployment

### PR DESCRIPTION
**Pull request by acidsennui**
Ref infrastructure PR https://github.com/aeternity/infrastructure/pull/29
Test build: https://circleci.com/gh/aeternity/epoch/4482
pivotal: https://www.pivotaltracker.com/story/show/155784885

### CODE REVIEW

**Comment by dincho** [comment]
@acidsennui Please rebase and fix the config file.

**Comment by dincho** [comment]
@acidsennui Please also add reference+link (repo) to the exact version of `deployment/ansible/inventory/ec2.py` in the file as comment

File: deployment/README.md [hunk 1]
```diff
@@ -2,12 +2,16 @@
...
-This implementation supports OpenStack dynamic inventory by using [Ansible 2.4 OpenStack inventory plugin](https://docs.ansible.com/ansible/devel/plugins/inventory/openstack.html).
+This implementation supports:
+
+OpenStack dynamic inventory by using [Ansible 2.4 OpenStack inventory plugin](https://docs.ansible.com/ansible/devel/plugins/inventory/openstack.html).
 To install it's dependencies with `pip`:
 ```bash
 pip install -r ansible/pip-requirements.txt
 ```
 
+AWS EC2 dynamic inventory [EC2 Inventory](https://github.com/ansible/ansible/blob/cdf8f29e70c15598d0a92c8f17841e58d240f94b/contrib/inventory/ec2.py)
```
Comment (dincho) [inline deployment/README.md]: This does not look good to me in the middle of nothing. Please see https://github.com/aeternity/epoch/blob/98c62239df9e6c88a3146a77134b431c3a676711/deployment/README.md

Plase take a time to better structure the documentation and split the supported inventories in separate sections.
Also the link should point to the [inventory documentation](https://docs.ansible.com/ansible/devel/plugins/inventory/aws_ec2.html) and the source version (tag) should be pointed in the ec2.py file e.g. https://github.com/ansible/ansible/blob/v2.5.0rc3/contrib/inventory/ec2.py

File: .circleci/config.yml [hunk 2]
```diff
@@ -384,6 +384,12 @@ jobs:
       - DEPLOY_ENV: uat
     steps: *deploy_steps
 
+  deploy_aws_dev:
```
Comment (dincho) [inline .circleci/config.yml]: Why do we keep this? I see it runs OK in the test builds already.

**Comment by dincho** [comment]
@acidsennui Please update your PR title/description with more meaningful information about the changes:

- references to the infra repo with related changes
- reference the PT ticket
- reference to successful test build

File: .circleci/config.yml (hunk 2 above)
Comment (acidsennui) [reply .circleci/config.yml]: @dincho not sure what do you mena, but you have there deploy_intenrgariton deploy_Dev1 etc... so i think we need one for deplou_aws_dev for specyfiint dev_aws env?

Comment (dincho) [reply .circleci/config.yml]: no, once we're ready we'll just tag AWS stack with `tag_env_uat` for example

File: deployment/README.md [hunk 3]
```diff
@@ -2,30 +2,68 @@
...
 ```bash
 source ~/my/secrets/openstack.rc
 ansible-inventory -i inventory/openstack.yml --list
 ```
 
-Make sure your OpenStack compute instances has correct "groups" metadata set.
-For example to add an instance to both groups "epoch" and "integration" use:
+#### Amazon Web Services
+
+Yous should make sure [AWS CommandLine interface credentials are set](http://docs.ansible.com/ansible/latest/intro_dynamic_inventory.html#example-aws-ec2-external-inventory-script) either by environment vairbales or ~/.aws/credentials file
```
Comment (happi) [inline deployment/README.md]: "You should"

Comment (dincho) [reply deployment/README.md]: fixed. thanks

File: deployment/README.md [hunk 4]
```diff
@@ -2,19 +2,42 @@
...
 ansible-inventory -i inventory/openstack.yml --list
 ```
+### Amazon Web Services
+Yous should make sure [AWS CommandLine interface credentials are set](http://docs.ansible.com/ansible/latest/intro_dynamic_inventory.html#example-aws-ec2-external-inventory-script) either by environment vairbales or ~/.aws/credentials file
+
+```bash
+ansible-inventory -i inventory/ec2.py --list
+```
+
+If you hace configured multiple AWS credentials you can pass AWS_PROFILE variable before command
```
Comment (lucafavatella) [inline deployment/README.md]: have

File: deployment/README.md [hunk 5]
```diff
@@ -2,19 +2,42 @@
...
+### Openstack
 You should make sure [OpenStack credentials are set](https://docs.openstack.org/python-openstackclient/latest/configuration/index.html#environment-variables)
 either by environment variables or clouds.yml file.
 
 ```bash
 source ~/my/secrets/openstack.rc
 ansible-inventory -i inventory/openstack.yml --list
 ```
+### Amazon Web Services
+Yous should make sure [AWS CommandLine interface credentials are set](http://docs.ansible.com/ansible/latest/intro_dynamic_inventory.html#example-aws-ec2-external-inventory-script) either by environment vairbales or ~/.aws/credentials file
```
Comment (lucafavatella) [inline deployment/README.md]: variables

**Comment by acidsennui** [comment]
Tags for aws env are here: https://github.com/aeternity/infrastructure/blob/bafbcdbfd383705481d7a1a8dbb33078fcd7736c/aws/ae-environment.yml#L81 it should not affect current deployment, just add new env in aws.

**Comment by dincho** [comment]
@lucafavatella This PR adds support to AWS dynamic inventory. As @acidsennui pointed (in addition to your finds) the AWS nodes will have a group `tag_env_uat` thus the current deployment should work as is, and deploy to the AWS nodes on tag.

**Comment by dincho** [comment]
@lucafavatella Once this and other infrastructure PR's for adding new nodes are merged, I'll do manual UAT (only AWS) deployment to ensure it works as expected and take the AWS UAT nodes in the testnet.

**Pull request merged**